libr/core/linux_heap_glibc.c

```diff
@@ -303,19 +336,45 @@ static void GH(print_arena_stats)(RCore *core, GHT m_arena, MallocState *main_ar
 }
 
 static bool GH(r_resolve_main_arena)(RCore *core, GHT *m_arena) {
-	if (!core || !core->dbg || !core->dbg->maps) {
-		return false;
+
+	r_return_val_if_fail (core && core->dbg && core->dbg->maps, false);
+
+	/*
+	if (core->dbg->main_arena_resolved) {
+		return true;
 	}
+	*/
 
+	const char *main_arena_str = "main_arena";
 	GHT brk_start = GHT_MAX, brk_end = GHT_MAX;
 	GHT libc_addr_sta = GHT_MAX, libc_addr_end = 0;
 	GHT addr_srch = GHT_MAX, heap_sz = GHT_MAX;
+	const char *libc_path = NULL;
+	GHT libc_addr = GHT_MAX;
+	GHT main_arena_sym = GHT_MAX;
 
 	if (r_config_get_i (core->config, "cfg.debug")) {
 		RListIter *iter;
 		RDebugMap *map;
 		r_debug_map_sync (core->dbg);
 		r_list_foreach (core->dbg->maps, iter, map) {
+			if (strstr (map->name, "/libc-") && map->perm == 4 && main_arena_sym == GHT_MAX) {
+				libc_path = map->name;
+				libc_addr = map->addr;
+				if (!libc_path) {
```

Comment:
if (!map->name) break..

make this conditional before the assignment

Reply:
Probably this check can be removed completely, as I think `map->name` should always be notnull.
There is a check in `r_debug_map_new` (though it misses the check of `map->name`).